crates/ruff/src/checkers/noqa.rs

```diff
@@ -161,11 +161,16 @@ pub fn check_noqa(
                                     Location::new(row + 1, start - leading_spaces),
                                     Location::new(row + 1, end + trailing_spaces),
                                 ));
-                            } else {
+                            } else if lines[row].chars().nth(end + 1).map_or(false, |c| c == '#') {
```

Comment:
Nit: I would find some comments at the top of each case explaining the case they handle and what fix they produce useful.
```suggestion
                            } 
                            // example: `# noqua # another comment`
                            else if lines[row].chars().nth(end + 1).map_or(false, |c| c == '#') {
```